### ISSUE TITLE
release binaries for apple silicon

### ISSUE THREAD

**Issue by juanfem** [mdavidsaver/p4p#88]
I recently got a MacBook with the M1 chip, and it would be nice to have the binaries for aarch64.

It is not a big issue, since pip can compile from sources and it works fine, but when building Docker images you need to find a way to make gcc and g++ available, at least during the build process.

Probably aarch64 binaries for epicscorelibs and pvxs would also be needed.

**Comment by mdavidsaver** [comment]
I agree that binaries for additional archs would be nice to have.  (eg. I make use of [piwheels](https://www.piwheels.org/project/p4p/) for ARMv7)

I have some [hint](https://github.com/dls-controls/cothread/issues/29) that recent OSX images + python releases will build with `-arch arm64`.

Do you know if wheels built like this on apple/x86_64 can be used on apple/arm64?

This will probably entail figuring out what the correct python wheel target name should be.  (something "universal2"?)  Currently the uploaded osx wheels are [named](https://pypi.org/project/p4p/#files) as `*-macosx_10_9_intel.whl` regardless of `-arch`.

Ideally, native arm64 builders seem like the way to go.  However, I don't think these are available at present from any of the free CI services.

**Comment by juanfem** [comment]
The wheel that would interest me the most is in fact `linux/aarch64`, which is the one I need to install on Docker images.

I tried building the wheel from a linux amd64 machine but I didn't succeed. I haven't even managed for the host machine, so I should keep trying... I tried to follow the pipeline you use on GHA.

**Comment by mdavidsaver** [comment]
linux/aarch64 would be of some interest to me as well.

My past attempts to cross-build `.whl` files have not be successful.  As I recall, the most success I had was using a mangled raspbian disk image, `qemu-arm-static`, and `systemd-nspawn`.  I could compile (very slowly!), but the tests failed as QEMU process emulation doesn't handle some of the obscure `ioctl()` calls that CA and PVA use.  I stopped at that point because the idea of trying to automate that process was... intimidating.
